From error27@gmail.com  Thu Jan 14 17:03:46 2010
From: Dan Carpenter <error27@gmail.com>
Date: Fri, 8 Jan 2010 14:51:50 +0300
Subject: Staging: pohmelfs: avoid null dereference
To: Evgeniy Polyakov <zbr@ioremap.net>
Cc: devel@driverdev.osuosl.org, Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <20100108115150.GA7840@bicker>

err_out_put is for when the netfs_trans_alloc() succeeded.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Acked-by: Evgeniy Polyakov <zbr@ioremap.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/pohmelfs/inode.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/pohmelfs/inode.c
+++ b/drivers/staging/pohmelfs/inode.c
@@ -322,7 +322,7 @@ int pohmelfs_write_create_inode(struct p
 	t = netfs_trans_alloc(psb, err + 1, 0, 0);
 	if (!t) {
 		err = -ENOMEM;
-		goto err_out_put;
+		goto err_out_exit;
 	}
 	t->complete = pohmelfs_write_inode_complete;
 	t->private = igrab(inode);
